schemas/theme/setting.json

```diff
@@ -42,127 +44,127 @@
   "$comment": "We're doing the weird allOf here so that we have better error messages when type is invalid. We're also doing the allOf so that we can use additionalProperties properly.",
   "allOf": [
     {
-      "if": { "properties": { "type": { "const": "header" } } },
+      "if": { "required": ["type"], "properties": { "type": { "const": "header" } } },
```

Comment:
Added the required type because I wanted to fallback to the generic docs when the rule doesn't match. 

When we omit the required, the rule matches when the type property isn't present. Which is not what we want to do.